core-build/gulp-core-build-typescript/src/ApiExtractorTask.ts

```diff
@@ -146,7 +146,10 @@ export class ApiExtractorTask extends GulpTask<IApiExtractorTaskConfig>  {
       };
 
       const extractor: Extractor = new Extractor(extractorConfig, extractorOptions);
-      extractor.analyzeProject();
+
+      // NOTE: processProject() returns false if errors or warnings occurred, however we
+      // already handle this above via our customLogger
+      extractor.processProject();
```

Comment:
how about adding if (!result) { completeCallback('extractor failed with n errors'); }?

Reply:
GCB already counts the number of errors and warnings that were passed to its custom logger and reports them like this:

```
Warning - tslint - src/Text.ts(41,12): error no-any: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type, the empty type ('{}'), or suppress this occurrence.
Warning - tslint - src/Text.ts(41,9): error no-unused-variable: 'x' is declared but never used.
Warning - tslint - src/Text.ts(41,9): error prefer-const: Identifier 'x' is never reassigned; use 'const' instead of 'let'.
Warning - [api-extractor] You have changed the public API signature for this project. Updating ../../common/reviews/api/sp-core-library.api.ts
Error - [api-extractor] src/Text.ts(10,1): A release tag (@alpha, @beta, @public, @internal) must be specified for Text
[15:23:14] Project @microsoft/sp-core-library version: 1.4.0-dev.14
[15:23:14] Build tools version: 3.2.7
[15:23:14] Node version: v6.10.2
[15:23:14] Total duration: 12 s
[15:23:14] Task warnings: 4
[15:23:14] Task errors: 1
```

Thus, the api-extractor task doesn't need to maintain its own statistics.

---
In reply to: [152714393](https://github.com/Microsoft/web-build-tools/pull/451#discussion_r152714393) [](ancestors = 152714393)